### ISSUE TITLE
Remove named function syntax from chrome *.js files

### ISSUE THREAD

**Issue by wolfbeast** [MoonchildProductions/UXP#516]
We're having a mix of named function syntax (old style) and non-named function syntax (new style) in our internal JavaScript.

The old style was mostly there for JS stack traces before they were as nice as what we have now. Since that's solved, having named functions has become unnecessary and will just take up JS data space when processing.

We should convert all named function syntax in JSO scope definitions to non-named function syntax in our internal scripts.
e.g.:
```javascript
WebConsoleFrame.prototype = {
    _initConnection: function WCF__initConnection() {
        ...
    },
   ...
}
```
becomes:
```javascript
WebConsoleFrame.prototype = {
    _initConnection: function() {
        ...
    },
   ...
}
```

This is very easy work, but should be taken in chunks to make reviewing easy and reduce the risk of bitrot while working on it. Made it a meta issue as a result. PRs for the various parts of our source tree can directly reference this issue, no need to create separate issues for solving parts of this.

**Comment by mattatobin** [comment]
Largely concentrated in Pale Moon and TychoAM plus stuff Mozilla hadn't gotten to at that point.

Was wondering about that so good to know.

**Comment by ThosRTanner** [comment]
Why use the `name : function(params)` syntax? eslint (for instance) has a rule that suggests just using `name(params)`

**Comment by mattatobin** [comment]
Because.

**Comment by ThosRTanner** [comment]
fair enough, just checking on the coding standards.

**Comment by mattatobin** [comment]
@ThosRTanner Why are you creating noise.. You do realize these functions are inside of objects right? They don't need an internal name for debugging purposes anymore.

**Comment by ThosRTanner** [comment]
Yes, I do realise these are in object. However javascript does allow na I merely wanted to check what your particular coding standards had to say about using `name() {` rather than `name: function() {`. This:

var drinks = {
    applejuice() { alert(...) },
    beer() { alert... }
}

is also permitted by the language, and in some places, encouraged as being more readable (YMMV which is why I'm asking).

**Comment by mattatobin** [comment]
Obviously, this issue states what we intend to do. Which is removed the named function syntax because it isn't needed as described in the opening description.

**Comment by wolfbeast** [comment]
Whether it is allowed or not doesn't matter. if they are inside an object the normal way to designate properties and functions alike inside them consistently is using `name :` and then whatever it is.
So in this case `name: function(params) { impl }` is what we want to use.

**Comment by wolfbeast** [comment]
How far are we with this? can this be closed?
